### PR TITLE
added a problem with solution on sliding window.

### PR DESCRIPTION
**Pull request by rohan472000**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sliding_window/minimum_size_subarray_sum.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    Returns the length of the shortest contiguous subarray
+     in nums whose sum is at least target.
+
+    Args:
+        target (int): The minimum sum that the subarray should have.
+        nums (list[int]): The array to search for subarrays in.
+
+    Returns:
+        int: The length of the shortest contiguous subarray
+        in nums whose sum is at least target. If no such subarray exists, returns 0.
```
[the remark this answers is not in the document]
Comment (rohan472000) [reply sliding_window/minimum_size_subarray_sum.py:12]: Sure...i will remove it

File: sliding_window/minimum_size_subarray_sum.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    Examples:
+        >>> minsubarraysum(7, [2, 3, 1, 2, 4, 3])
+        2
```
[the remark this answers is not in the document]
Comment (rohan472000) [reply sliding_window/minimum_size_subarray_sum.py:16]: Yeah I will do it sooner

[the remark this answers is not in the document]
Comment (rohan472000) [reply sliding_window/minimum_size_subarray_sum.py:8]: @CaedenPH, done....

File: sliding_window/minimum_size_subarray_sum.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rohan472000) [reply sliding_window/minimum_size_subarray_sum.py]: Done

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
It's probably best to place this in an existing directory rather than creating a new directory just for this file

**Comment by rohan472000** [comment]
@tianyizheng02, well there is no directory for this algo, so i made it where all questions and answers based on sliding window can be kept.

**Comment by tianyizheng02** [comment]
Algorithms in this repo are placed in directories based on broad categories like dynamic programming, machine learning, sorting algorithms, etc. It's best to place new algorithm files in the same directory as similar algorithms rather than making a whole new directory for a single algorithm.

**Comment by rohan472000** [comment]
yeah got it....made the changes

**Comment by rohan472000** [comment]
@tianyizheng02 , why ruff is failed when I changed the directory?

**Comment by rohan472000** [comment]
@tianyizheng02 ,  merge this if u have reviewed..

**Comment by rohan472000** [comment]
@cclauss , please review and if changes needed then suggest.

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        0
+        >>> minsubarraysum(10, [10, 20, 30])
+        1
+        >>> minsubarraysum(7, [1, 1, 1, 1, 1, 1, 10])
+        1
+    """
+
+    if len(numbers) == 0:
+        return 0
+
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py:27]: Please deal similarly with all falsy values: False, None, "", [], {}, 0, 0.0, (, ), etc.
```suggestion
    if not numbers:
        return 0
```

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    left = right = curr_sum = 0
+    min_len = float("inf")
+
+    while right < len(numbers):
+        curr_sum += numbers[right]
+        while curr_sum >= target:
+            min_len = min(min_len, right - left + 1)
+            curr_sum -= numbers[left]
+            left += 1
+        right += 1
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py:37]:
```suggestion
    for right, number in enumerate(numbers):
        curr_sum += number
        while curr_sum >= target:
            min_len = min(min_len, right - left + 1)
            curr_sum -= numbers[left]
            left += 1
```

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        1
+        >>> minsubarraysum(7, [1, 1, 1, 1, 1, 1, 10])
+        1
+    """
+
+    if len(numbers) == 0:
+        return 0
+
+    left = right = curr_sum = 0
+    min_len = float("inf")
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py:29]: Let's keep the length an integer.
```suggestion
    min_len = sys.maxsize
```
https://docs.python.org/3/library/sys.html#sys.maxsize

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    while right < len(numbers):
+        curr_sum += numbers[right]
+        while curr_sum >= target:
+            min_len = min(min_len, right - left + 1)
+            curr_sum -= numbers[left]
+            left += 1
+        right += 1
+
+    return int(min_len) if min_len != float("inf") else 0
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py:39]:
```suggestion
    return 0 if min_len == sys.maxsize else min_len
```

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+def minsubarraysum(target: int, numbers: list[int]) -> int:
+    """
+    Returns the length of the shortest contiguous subarray
+     in nums whose sum is at least target.
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py:4]:
```suggestion
     in a list of numbers whose sum is at least target.
```

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
+def minsubarraysum(target: int, numbers: list[int]) -> int:
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py:1]:
```suggestion
def minimum_subarray_sum(target: int, numbers: list[int]) -> int:
```

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        2
+        >>> minsubarraysum(7, [2, 3, -1, 2, 4, -3])
+        4
+        >>> minsubarraysum(11, [1, 1, 1, 1, 1, 1, 1, 1])
+        0
+        >>> minsubarraysum(10, [1, 2, 3, 4, 5, 6, 7])
+        2
+        >>> minsubarraysum(5, [1, 1, 1, 1, 1, 5])
+        1
+        >>> minsubarraysum(0, [])
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py:17]: Add tests for 0, [1, 2, 3] and for -6, []

**Review by cclauss — CHANGES_REQUESTED** [review]

File: dynamic_programming/minimum_size_subarray_sum.py (hunk 9 above)
Comment (rohan472000) [reply dynamic_programming/minimum_size_subarray_sum.py]: added.

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        0
+        >>> minsubarraysum(10, [10, 20, 30])
+        1
+        >>> minsubarraysum(7, [1, 1, 1, 1, 1, 1, 10])
+        1
```
[the remark this answers is not in the document]
Comment (rohan472000) [reply dynamic_programming/minimum_size_subarray_sum.py]: i didn't get that..please check what i have committed now

File: dynamic_programming/minimum_size_subarray_sum.py [hunk 11]
```diff
@@ -59,7 +59,4 @@ def minimum_subarray_sum(target: int, numbers: list[int]) -> int:
     return 0 if min_len == sys.maxsize else min_len
 
 
-if __name__ == "__main__":
```
Comment (cclauss) [inline dynamic_programming/minimum_size_subarray_sum.py]: Why this change?

Comment (rohan472000) [reply dynamic_programming/minimum_size_subarray_sum.py]: now I removed it

**Comment by cclauss** [comment]
Why can't we have checkboxes like every other pull request has?
image

**Comment by rohan472000** [comment]
yeah ...filled the checkboxes

**Comment by cclauss** [comment]
Markdown has a syntax for checklists... https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists#creating-task-lists

**Comment by tianyizheng02** [comment]
No, you fill the checkboxes like this:
```
* [x] Test 1
* [ ] Test 2
* [ ] Test 3
```
And it should look like this in the preview:
* [x] Test 1
* [ ] Test 2
* [ ] Test 3

**Comment by rohan472000** [comment]
> No, you fill the checkboxes like this:
> 
> ```
> * [x] Test 1
> * [ ] Test 2
> * [ ] Test 3
> ```
> 
> And it should look like this in the preview:
> 
> * [x]  Test 1
> * [ ]  Test 2
> * [ ]  Test 3

Thanks....I changed it.

**Comment by rohan472000** [comment]
@cclauss @tianyizheng02 
I'm getting below error, what case can we add to solve these testcases?


Failure
<Click to see difference>

**********************************************************************
File "C:\pycharm\codes\Python-pull\sliding_window\minimum_size_subarray_sum.py", line 20, in minimum_size_subarray_sum.minimum_subarray_sum
Failed example:
    minimum_subarray_sum(0, [1, 2, 3])
Expected:
    0
Got:
    1



Failure
<Click to see difference>

**********************************************************************
File "C:\pycharm\codes\Python-pull\sliding_window\minimum_size_subarray_sum.py", line 32, in minimum_size_subarray_sum.minimum_subarray_sum
Failed example:
    minimum_subarray_sum(-6, [3, 4, 5])
Expected:
    0
Got:
    1

**Comment by tianyizheng02** [comment]
> Returns the length of the shortest contiguous subarray in a list of numbers whose sum is at least target.

So then `minimum_subarray_sum(0, [1, 2, 3])` _should_ return 1, not 0 like you wrote, because the shortest contiguous subarray with sum ≥ 0 is [1].

Likewise, `minimum_subarray_sum(-6, [3, 4, 5])` _should_ return 1, not 0 like you wrote, because the shortest contiguous subarray with sum ≥ -6 is [3].

In short, the code isn't wrong; it's those two tests that are wrong. Adding more test cases won't fix the issue.

**Comment by rohan472000** [comment]
done...i was thinking to do that earlier...

**Comment by rohan472000** [comment]
@cclauss , please review the code.

**Comment by rohan472000** [comment]
@cclauss ,It has been more than 2 weeks of this PR, review it, if something has to be changed in it then suggest.
